src/test/java/org/hydev/mcpm/client/DownloaderTest.java

```diff
@@ -0,0 +1,34 @@
+package org.hydev.mcpm.client;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hydev.mcpm.client.database.MirrorSelector.MIRROR_LIST_URL;
+import static org.hydev.mcpm.utils.NetworkUtils.ping;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+class DownloaderTest
+{
+    static boolean hasInternet = ping(MIRROR_LIST_URL) != -1;
+
+    @Test
+    void test()
+    {
+        assumeTrue(hasInternet);
+        String link = "https://mcpm.hydev.org/db";
+        File out = new File("./build/db.json");
+        String link1 = "https://mcpm.hydev.org/db.zst";
+        File out1 = new File("./build/db.zst");
+        Downloader downloader = new Downloader().showProgress(true).threads(2);
```

Comment:
This is an integration test and I would advise against including it. Can you put these integration tests outside of the regular test folder (under another target)?

Reply:
hmmm it should be fine since I assumed `hasInternet` (if the assumption is false then the test wouldn't run)

do you think this is ok? if not maybe I can make this into a main method so that it doesn't run as a unit test

Reply:
Yeah a main method would be perfect. You aren't making any assertions here so I think it makes the most sense.

`hasInternet` is a fine assumption but I don't think it removes all that uncertainty that comes with making a net request. The test can still behave differently on different connections at different times of the day (and as we've seen Bahen internet can cause this stuff to fail).

Reply:
ok removed